File: app/src/main/java/za/co/duartbreedt/androidconceptsreference/pusheen/PusheenViewHolder.java
package za.co.duartbreedt.androidconceptsreference.pusheen;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import za.co.duartbreedt.androidconceptsreference.databinding.ViewItemPusheenBinding;

public class PusheenViewHolder extends RecyclerView.ViewHolder {

    private final ViewItemPusheenBinding binding;

    public PusheenViewHolder(@NonNull ViewItemPusheenBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void setPusheen(String pusheen) {
        binding.textPrimary.setText(pusheen);
    }
}

File: app/src/main/java/za/co/duartbreedt/androidconceptsreference/pusheen/PusheenViewHolder.kt
package za.co.duartbreedt.androidconceptsreference.pusheen

import androidx.recyclerview.widget.RecyclerView
import za.co.duartbreedt.androidconceptsreference.databinding.ViewItemPusheenBinding

class PusheenViewHolder(
    private val binding: ViewItemPusheenBinding
) : RecyclerView.ViewHolder(binding.root) {

    fun setPusheen(pusheen: String?) {
        binding.textPrimary.text = pusheen
    }
}
